azure_function/resource-graph-collector/__init__.py

```diff
@@ -44,19 +48,34 @@ def main(event: func.TimerRequest) -> None:
         logger.debug(f"Query results dump:\n---\n{result.as_dict()}\n---")
 
         # publish to loki
-        loki_logger = LokiPublisher(loki_endpoint=os.environ['LOKI_ENDPOINT'],
-                                    auth=(os.environ['LOKI_USERNAME'], os.environ['LOKI_PASSWORD']),
-                                    tags={'inventory_type': 'reference_architecture',
-                                        'graph_query_name': resource_graph_query.name})
+        loki_logger = LokiPublisher(
+            loki_endpoint=os.environ["LOKI_ENDPOINT"],
+            auth=(os.environ["LOKI_USERNAME"], os.environ["LOKI_PASSWORD"]),
+            tags={
+                "inventory_type": "reference_architecture",
+                "graph_query_name": resource_graph_query.name,
+            },
+        )
 
-        fields_to_labels_string = os.environ.get('LOKI_LABEL_NAMES', None)
+        fields_to_labels_string = os.environ.get("LOKI_LABEL_NAMES", None)
         fields_to_labels = []
         if fields_to_labels_string:
-            fields_to_labels = [e.strip() for e in fields_to_labels_string.split(',')]
+            fields_to_labels = [e.strip() for e in fields_to_labels_string.split(",")]
 
         if len(fields_to_labels) > MAX_LABELS:
-            raise ValueError(f"A maximum of {MAX_LABELS} is supported. You requested: {len(fields_to_labels)} labels, "
-                            f"namely: [{', '.join(fields_to_labels)}]")
+            raise ValueError(
+                f"A maximum of {MAX_LABELS} is supported. You requested: {len(fields_to_labels)} labels, "
+                f"namely: [{', '.join(fields_to_labels)}]"
+            )
 
-        for item in result.as_dict().get('data', []):
+        for item in result.as_dict().get("data", []):
             loki_logger.publish(item, fields_to_labels)
+
+        # publish to Azure Blob container
+        azure_publisher = AzurePublisher(
+            connection_string=os.environ["STORAGE_ACCOUNT_CONNECTION"],
+            container_name=os.environ["CONTAINER_NAME"],
+        )
+        azure_publisher.publish(
+            name=resource_graph_query.name, data=result.as_dict().get("data", [])
+        )
```

Comment:
could you add this behind a feature switch (env var is ok)? so it's backward compatible

Reply:
Great idea, I've push a new proposal. Looking forwarding hearing from you.